include/zephyr/drivers/sensor/tcs3400.h

```diff
@@ -12,6 +12,11 @@
 enum sensor_attribute_tcs3400 {
 	/** RGBC Integration Cycles */
 	SENSOR_ATTR_TCS3400_INTEGRATION_CYCLES = SENSOR_ATTR_PRIV_START,
+	/**
+	 * If set the clear channel reports the measurement from the IR
+	 * sensor.
+	 */
+	SENSOR_ATTR_TCS3400_IR_ENABLE,
```

Comment:
I was actually thinking that you would expose IR through the SENSOR_CHAN_IR channel? i.e fetching would always fetch the 5 five channels, flipping the IR bit between two reads of the IR/IC channel. But I haven't looked at the datasheet closely to see if that would require extra waits and stuff, or if it would make the sampling take much longer.

Reply:
No the headline is somewhat misleading :-) there's 5 channels but 4 ADC, so you always get 4 output and the clear one can go to either diode, go figure. Basically it's just one bit that controls the switch in the diagram here.

![x](https://github.com/zephyrproject-rtos/zephyr/assets/891546/c1cb0903-c6a6-4826-8be2-c644d73e6604)

Reply:
Ya but from an end-user perspective wouldn't you want to expose it as 5 actual channels? I don't think it should matter how the vendor decided to share the same ADC and registers for Clear and IR (like if they had only one for everything, you would still want to expose things using Zephyr's nice channels right?).
Sample fetching would end up storing data in a `uint16_t sample_crgbi[5]`, reading the first 4 channels like it 's done now, flip the bit, and then read `CDATAL` and `CDATAH` only to get the missing 5th?

Reply:
Oh I see what you mean. Yeah it's certainly an option but I'd rather not redo the whole fetching process twice all the time, it already involves many i2c transactions as it is and this would add one more and double up the whole story. Our application does not even use that feature right now so I don't even have a solid use case. I think I'd rather just expose the internal setting for now and it can be revisited if we get a better idea what to do with it.

Reply:
I see. In that case I'm actually not sure it's worth the change, since it feels odd to not expose IR through the dedicated channel. At least if you want to keep things mutually exclusive for now, could you add the required logic in the switch-case so that one can access IR through `SENSOR_CHAN_IR` if they've set the option (and sensor clear would -ENOTSUP out, and vice-versa if they've enabled the clear channel)

Reply:
Fair enough, let's rework it when we have a solid use case. :-)

Reply:
I'm sorry I blogged about it! 🙈

Reply:
Ah it was a good call :-)